# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1502900298 18000
#      Wed Aug 16 11:18:18 2017 -0500
# Node ID fe780a4b89c1c184500ac29fd604703850d3dc12
# Parent  4be7a254f7521e308c9f7dc1ac09c953a6300e47
Supply missing axis labels for gnuplot toolkit (bug #51759).

* __gnuplot_draw_axes__.m (do_tics_1): Assign empty string to 'labels' if
  'labels' is empty (i.e., size 0x0).  Use repmat() to repeat labels when the
  number of labels is shorter than number of tics.  Remove the conditional test
  that resets index k to the start of labels when there are too few.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -2325,28 +2325,26 @@ function do_tics_1 (ticmode, tics, mtics
     fprintf (plot_stream, " %.15g) %s;\n", tics(end), fontspec);
   else
     fprintf (plot_stream, "set %stics %s %s %s %s %s %s;\n", ax,
              tickdir, ticklength, axispos, mirror, colorspec, fontspec);
   endif
   if (strcmp (labelmode, "manual"))
     k = 1;
     ntics = numel (tics);
-    nlabels = numel (labels);
+    labels(end+1:1) = {""};
+    labels = repmat (labels(:), ceil (ntics/numel (labels)), 1);
     fprintf (plot_stream, "set %stics add %s %s %s %s (", ax,
              tickdir, ticklength, axispos, mirror);
     labels = strrep (labels, "%", "%%");
     for i = 1:ntics
       fprintf (plot_stream, " \"%s\" %.15g", labels{k++}, tics(i));
       if (i < ntics)
         fputs (plot_stream, ", ");
       endif
-      if (k > nlabels)
-        k = 1;
-      endif
     endfor
     fprintf (plot_stream, ") %s %s;\n", colorspec, fontspec);
   endif
   if (strcmp (mtics, "on"))
     fprintf (plot_stream, "set m%stics %d;\n", ax, num_mtics);
   else
     fprintf (plot_stream, "unset m%stics;\n", ax);
   endif
